server/src/main/java/com/google/sps/utils/FirebaseAdReader.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.utils;
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.sps.utils.Ad;
+import com.google.sps.utils.AdRowProcessor;
+import com.google.sps.utils.FirebaseAdReader;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.concurrent.TimeUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Description: Utility class that reads ad objects from Firestore. 
+ * Author: Kira Toal
+ * Date: July 14, 2020
+ */ 
+public final class FirebaseAdReader {
+
+  public static List<Ad> readAds(String collection, String databaseURL, String pathToServiceAccount,
+    int limit, int timeoutAllowance) throws Exception {
```

Comment:
Since timeoutAllowance is in seconds, just name it timeoutSecs.  Allowance is redundant anyway.